.github/workflows/ci.yml

```diff
@@ -9,12 +9,26 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - name: Check out
+        uses: actions/checkout@v3
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: nightly-2022-11-03
-      - uses: Swatinem/rust-cache@v2
-      - run: make install
-      - run: make test
-      - run: make check-format
\ No newline at end of file
+      - name: Set up cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
hash of cargo.lock will stay the same even if `starkware-libs/cairo` is updated with new commits no?

Reply:
I think it will, but the next step will ensure to pull al changes and only then apply those